__tests__/app.test.js

```diff
@@ -0,0 +1,52 @@
+const request = require('supertest')
+let db = require('../db/connection')
+let app = require('../app.js')
+let seed = require('../db/seeds/seed')
+
+// bring in data from test data files
+let testData = require('../db/data/test-data/index.js')
+
+// re-seed database before each test
+beforeEach(() => {
+    return seed(testData);
+})
+
+// after all tests, disconnect from database
+afterAll(() => {
+    return db.end();
+})
+
+// main testing body
+
+describe('GET /api/categories',() => {
+    test("check that status 200 is returned",() => {
+        return request(app).get('/api/categories').expect(200)
+    })
+    test("check that 4 rows are returned and they have the required format",() => {
+        return request(app).get('/api/categories')
+        .then(({body: { categories }}) => {
+            expect(categories).toHaveLength(4);
+            categories.forEach((category) => {
+                expect(category).toEqual(
+                    expect.objectContaining({
+                        slug: expect.any(String),
+                        description: expect.any(String)
+                    })
+                )
+            })
+
+        })
+    })
+})
+
+describe('GET /api/categorys',() => {
```

Comment:
Good to have generic 404 error handling, but I would say that this isn't relevant to this specific ticket. Try and only write relevant code.